### PR TITLE
Patch 1

### PR DESCRIPTION
**Pull request by addy1997**
created a code for catalan number

### CODE REVIEW

**Review by Mystigan — CHANGES_REQUESTED** [review]
- Please ensure the program is logically correct. Also add proper unit tests. This ensures correctness to a great extent.
- Remove main function.
- Refactor with proper package and function names.
- Why did you delete the `sorts_test.go` file? Please add it back, it is an important part of the sorts package.

**Comment by addy1997** [comment]
@Mystigan I have included the test and renamed the functions [#239 ]

File: other/catalan.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package testing101
```
Comment (Mystigan) [inline other/catalan.go:1]: When I said proper package names, I meant it in the context of the project. The Go way to do it is that the package name is the folder in which the file is kept.

File: other/catalan_test.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// reference for testing https://www.calhoun.io/how-to-test-with-go/
+package testing101
+
+import (
+
+  "testing"
+)
+
+func TestCatalan(t *testing.T){
+
```
Comment (Mystigan) [inline other/catalan_test.go:10]: - Add more tests.
- Use table driven tests.
- Enclose test cases in t.Run().

**Review by Mystigan — CHANGES_REQUESTED** [review]
- Look at inline comments.
- Add the sorts_test file back. Why did you delete it?
- Remove extra whitespaces.

**Comment by tjgurwara99** [comment]
Please take a look at our new [contribution guidelines](https://github.com/TheAlgorithms/Go/blob/d2f12849df76203d80762ceca5105531d3424384/CONTRIBUTING.md) and make changes accordingly. Currently this PR is not up to standard. If no changes are made in the next two weeks, I will consider this PR to be stale and will close this PR then.

**Comment by addy1997** [comment]
> Please take a look at our new [contribution guidelines](https://github.com/TheAlgorithms/Go/blob/d2f12849df76203d80762ceca5105531d3424384/CONTRIBUTING.md) and make changes accordingly. Currently this PR is not up to standard. If no changes are made in the next two weeks, I will consider this PR to be stale and will close this PR then.

Wow! that's a quick response. 
Add this point to your guidelines
*Respond to the PR before it becomes too stale*

**Comment by tjgurwara99** [comment]
Unfortunately, I became a maintainer recently so your point is valid but I'm not at fault. Hope you enjoy the rest of your day.

**Comment by siriak** [comment]
@addy1997 I'm sorry that it took you so long to get meaningful feedback, we're working on that project in our spare time and cannot allocate more resources to that activity. We'll do our best to review new contributions as soon as possible in the future. Have a good day!

**Pull request closed without merging**